lib/gems/pending/util/object_storage/miq_swift_storage.rb

```diff
@@ -0,0 +1,164 @@
+  end
+
+  def uri_to_object_path(remote_file)
+    # Strip off the leading "swift://" and the container name from the URI"
+    # Also remove the leading delimiter.
+    object_file_with_bucket = URI.split(URI.encode(remote_file))[5]
+    object_file_with_bucket.split(File::Separator)[2..-1].join(File::Separator)
+  end
+
+  def upload_single(dest_uri)
+    #
+    # Get the remote path, and parse out the bucket name.
+    #
+    object_file = uri_to_object_path(dest_uri)
+    #
+    # write dump file to swift
+    #
+    logger.debug("Writing [#{source_input}] to => Bucket [#{container_name}] using object file name [#{object_file}]")
+    begin
+      swift_file = container.files.new(:key => object_file)
+      params     = {
+        :expects       => [201, 202],
+        :headers       => {},
+        :request_block => -> { read_single_chunk },
+        :idempotent    => false,
+        :method        => "PUT",
+        :path          => "#{Fog::OpenStack.escape(swift_file.directory.key)}/#{Fog::OpenStack.escape(swift_file.key)}"
+      }
+      #
+      # Because of how `Fog::OpenStack` (and probably `Fog::Core`) is designed,
+      # it has hidden the functionality to provide a block for streaming uploads
+      # that is available out of the box with Excon.
+      #
+      # we use .send here because #request is private
+      # we can't use #put_object (public) directly because it doesn't allow a 202 response code,
+      # which is what swift responds with when we pass it the :request_block
+      # (This allows us to stream the response in chunks)
+      #
+      swift_file.service.send(:request, params)
+      clear_split_vars
+    rescue Excon::Errors::Unauthorized => err
+      msg = "Access to Swift container #{@container_name} failed due to a bad username or password. #{err}"
+      logger.error(msg)
+      raise err, msg, err.backtrace
+    rescue => err
+      msg = "Error uploading #{source_input} to Swift container #{@container_name}. #{err}"
+      logger.error(msg)
+      raise err, msg, err.backtrace
+    end
+  end
+
+  def mkdir(_dir)
+    container
+  end
+
+  #
+  # Some calls to Fog::Storage::OpenStack::Directories#get will
+  # return 'nil', and not return an error.  This would cause errors down the
+  # line in '#upload' or '#download'.
+  #
+  # Instead of investigating further, we created a new method that is in charge of
+  # OpenStack container creation, '#create_container', and that is called from '#container'
+  # if 'nil' is returned from 'swift.directories.get(container_name)', or in the rescue case
+  # for 'NotFound' to cover that scenario as well
+  #
+
+  def container(create_if_missing = true)
+    @container ||= begin
+                     container = swift.directories.get(container_name)
+                     logger.debug("Swift container [#{container}] found") if container
+                     unless container raise Fog::Storage::OpenStack::NotFound
```

Comment:
This is `C` style syntax for conditional, and is causing your syntax errors.  Should be:

```ruby
raise Fog::Storage::OpenStack::NotFound unless container
```

Reply:
Yeah I already caught this with a test I was trying.  Thanks.